### DISCUSSION TITLE
Is it possible to programmatically change flag value? For testing different codepaths?

### DISCUSSION THREAD

**Discussion by paoloyx** [Flagsmith/flagsmith#282] [General] [answered]
Hi all,

I'm looking after Flagsmith and I find it very powerful, so far. There's one thing that I'm not able to understand, though: how can I manipulate flags values during unit tests?

When testing my code I'd like to test all paths, with flag enabled and/or disabled.

How should I do it using Flagsmith?

**Comment by dabeeeenster** [comment, marked as answer]
Hi @paoloyx  - thanks for the question - I've moved it to our Discussion area as its more relevant here. 

You can programatically drive Flagsmith via our API - [more details here](https://docs.flagsmith.com/advanced-use/api). Some of our users use the API to drive tests by modifying the Flagsmith flags for a particular test environment as part of the tests themselves. 

For particular languages, it would depend on what the most idiomatic way of implementing these code path changes would be - and that would depend on the language at hand.

**Reply by paoloyx** [reply]
@dabeeeenster Thank you! Ok, that's pretty clear.
